filters/scheduler/lifo.go

```diff
@@ -0,0 +1,335 @@
+		}
+	}
+
+	if len(args) > 3 {
+		d, err := durationArg(args[3])
+		if err != nil {
+			return nil, err
+		}
+		if d >= 1*time.Millisecond {
+			l.config.Timeout = d
+		}
+	}
+
+	return l, nil
+}
+
+// Config returns the scheduler configuration for the given filter
+func (l *lifoFilter) Config(*scheduler.Registry) scheduler.Config {
+	return l.config
+}
+
+// SetStack binds the stack to the current filter context
+func (l *lifoFilter) SetStack(s *scheduler.Stack) {
+	l.stack = s
+}
+
+// GetStack is only used in tests
+func (l *lifoFilter) GetStack() *scheduler.Stack {
+	return l.stack
+}
+
+// Key returns the scheduler string
+func (l *lifoFilter) Key() string {
+	return l.key
+}
+
+// SetKey sets the scheduler string, which should be called from the
+// PostProcessor
+func (l *lifoFilter) SetKey(k string) {
+	l.key = k
+}
+
+// Request is the filter.Filter interface implementation. Request will
+// increase the number of inflight requests and respond to the caller,
+// if the bounded stack returns an error. Status code by Error:
+//
+// - 503 if jobstack.ErrStackFull
+// - 502 if jobstack.ErrTimeout
+func (l *lifoFilter) Request(ctx filters.FilterContext) {
+	request(l.GetStack(), lifoKey, ctx)
+}
+
+// Response is the filter.Filter interface implementation. Response
+// will decrease the number of inflight requests.
+func (l *lifoFilter) Response(ctx filters.FilterContext) {
+	response(lifoKey, ctx)
+}
+
+// Config returns the scheduler configuration for the given filter
+func (l *lifoGroupFilter) Config(r *scheduler.Registry) scheduler.Config {
+	return r.Config(l.key)
+}
+
+// SetStack binds the stack to the current filter context
+func (l *lifoGroupFilter) SetStack(s *scheduler.Stack) {
+	l.stack = s
+}
+
+// GetStack is only used in tests
+func (l *lifoGroupFilter) GetStack() *scheduler.Stack {
+	return l.stack
+}
+
+// Key returns the scheduler group string
+func (l *lifoGroupFilter) Key() string {
+	return l.key
+}
+
+// SetKey is a noop to implement the LIFOFilter interface
+func (*lifoGroupFilter) SetKey(string) {}
+
+// Request is the filter.Filter interface implementation. Request will
+// increase the number of inflight requests and respond to the caller,
+// if the bounded stack returns an error. Status code by Error:
+//
+// - 503 if jobstack.ErrStackFull
+// - 502 if jobstack.ErrTimeout
+func (l *lifoGroupFilter) Request(ctx filters.FilterContext) {
+	request(l.GetStack(), lifoGroupKey, ctx)
+}
+
+// Response is the filter.Filter interface implementation. Response
+// will decrease the number of inflight requests.
+func (l *lifoGroupFilter) Response(ctx filters.FilterContext) {
+	response(lifoGroupKey, ctx)
+}
+
+func request(s *scheduler.Stack, key string, ctx filters.FilterContext) {
+	if s == nil {
+		log.Warningf("Unexpected scheduler.Stack is nil for key %s", key)
+		return
+	}
+
+	done, err := s.Ready()
+	if err != nil {
+		// TODO:
+		// - replace the log with metrics
+		// - allow custom status code
```

Comment:
@aryszka I added a comment in the user docs and godoc of this package, is this good enough or do you have any other recommendations before merging this PR?

Reply:
I think we can go like this in the first version. We can drop the todo.